sycl/include/sycl/ext/oneapi/matrix/matrix-unified-utils.hpp

```diff
@@ -68,7 +68,7 @@ convertMatrixUseStringToEnum(const char *UseString) {
   return std::nullopt;
 }
 
-inline __SYCL_ALWAYS_INLINE __spv::MatrixLayout joint_matrix_layout_to_spv(
+constexpr inline __spv::MatrixLayout joint_matrix_layout_to_spv(
     sycl::ext::oneapi::experimental::matrix::layout Layout) {
```

Comment:
I guess it was not possible to capture and preserve the layout as constant at this stage?

Reply:
Compiler considers it to be a runtime value as `joint_matrix_load/store` functions (that calls `joint_matrix_layout_to_spv`) can not be in `constexpr` context, as they call external functions (`__spirv_CooperativeMatrixLoad/StoreKHR`). I'm leaving `constexpr` keyword here as it's really our intention, yet not enforced by the compiler.